ksql-serde/src/main/java/io/confluent/ksql/serde/Format.java

```diff
@@ -107,6 +86,16 @@ public void validateProperties(final Map<String, String> properties) {
     });
   }
 
+  /**
+   * If the format accepts custom properties in the WITH clause of the statement,
+   * this method dictates which property keys are valid keys. This is used in
+   * conjunction with {@link #validateProperties(Map)} to ensure that the format
+   * configuration is valid.
+   *
+   * @return a set of valid property names
+   */
+  Set<String> getSupportedProperties();
```

Comment:
default to empty set?

Reply:
I was debating this or forcing people to implement it - I'm happy defaulting to empty set.